metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/ESIndexBuilder.java

```diff
@@ -89,9 +91,33 @@ public void buildIndex(String indexName, Map<String, Object> mappings, Map<Strin
     String tempIndexName = indexName + "_" + System.currentTimeMillis();
     createIndex(tempIndexName, mappings, finalSettings);
     try {
-      searchClient.reindex(
-          new ReindexRequest().setSourceIndices(indexName).setDestIndex(tempIndexName),
-          RequestOptions.DEFAULT);
+      TaskSubmissionResponse reindexTask;
```

Comment:
Since this is a big chunk of logic, i'd prefer if we factored it into a separate, appropriately-named method